### ISSUE TITLE
Dialogs should not assume to be alerts

### ISSUE THREAD

**Issue by steverep** [material-components/material-web#3095]
the `mwc-dialog` component currently hard codes:
- `role="alertdialog"`
- `aria-describedby="content"`

This only makes sense for dialogs that meet the extra qualifications for an ARIA alert, but most dialogs do not and the component is much more versatile than that.

I recommend making the following changes:
1. Give the implementer the choice between roles `dialog` or `alertdialog`, and the default should be `dialog` to be least intrusive to assistive technology
2. Set `aria-describedby` to all contents only when the role is `alertdialog` as the contents can be assumed to be concise
3. For the `dialog` role, give the implementer some control over `aria-describedby`, and omit it by default to be least disruptive to assistive technology

**Comment by asyncLiz** [comment]
Fixed in M3, dialog's are `dialog` role by default and can be set to an `alertdialog` with the `type="alert"` attribute
